src/linearoperators/DFT.jl

```diff
@@ -70,102 +71,115 @@ julia> A*ones(3)
 struct IDFT{N,
 	       C<:RealOrComplex,
 	       D<:RealOrComplex,
+		   E,
 	       T1<:AbstractFFTs.Plan,
-	       T2<:AbstractFFTs.Plan} <: FourierTransform{N,C,D,T1,T2}
+	       T2<:AbstractFFTs.Plan} <: FourierTransform{N,C,D,E,T1,T2}
 	dim_in::NTuple{N,Int}
 	A::T1
 	At::T2
 end
 
 # Constructors
 #standard constructor
-DFT(dim_in::NTuple{N,Int}) where {N} = DFT(zeros(dim_in))
-
-function DFT(x::AbstractArray{D,N}) where {N,D<:Real}
-	A,At = plan_fft(x), plan_bfft(fft(x))
-	DFT{N,Complex{D},D,typeof(A),typeof(At)}(size(x),A,At)
+DFT(dim_in::NTuple{N,Int},dims=1:N) where {N} = DFT(zeros(dim_in),dims)
+
+function DFT(x::AbstractArray{D,N},dims=1:ndims(x)) where {N,D<:Real}
+	y2 = zeros(complex(D),size(x))
+	A = plan_fft(x,dims)
+	At = plan_bfft(y2,dims)
+	dim_in = size(x)
+	DFT{N,Complex{D},D,dims,typeof(A),typeof(At)}(dim_in,A,At)
```

Comment:
and hence
```julia
d = prod(size(x,e) for e in dims)
DFT{N,Complex{D},D,typeof(d),dims,typeof(A),typeof(At)}(dim_in,A,At,d)
```

Reply:
also are you sure of `d = prod(size(x,e) for e in dims)`?

Reply:
Makes sense to do the normalization here. But just to make sure, the same normalization also applies to `mul!`:
https://github.com/kul-forbes/AbstractOperators.jl/blob/e16bbe48e82c334e9435b6692e621f72a73373cd/src/linearoperators/DFT.jl#L155-L160

Are we certain that `x` from the constructor and `b` from `mul!` are the same dimensions? I guess the input `b::AbstractArray{C,N}` suggests so...

Reply:
Quite certain about `d = prod(size(x,e) for e in dims)` but will try to add a more convincing comment later :)